packages/x-data-grid/src/components/GridHeader.tsx

```diff
@@ -8,7 +8,12 @@ export function GridHeader() {
   return (
     <React.Fragment>
       <GridPreferencesPanel />
-      {rootProps.slots.toolbar && <rootProps.slots.toolbar {...rootProps.slotProps?.toolbar} />}
+      {rootProps.slots.toolbar && (
+        <rootProps.slots.toolbar
+          // Fixes error augmentation issue https://github.com/mui/mui-x/pull/15255#issuecomment-2454721612
+          {...(rootProps.slotProps?.toolbar as any)}
```

Comment:
I did this to make the types work, I think it's an acceptable solution as it's internal only.

Reply:
I'm fine if you are happy with this solution. 👌 
The component is public, but do you mean that if someone wants to override this slot, they will create their component?

Reply:
No, I meant that `as any` above doesn't impact the users passing the slot props